engine/gamesys/src/gamesys/scripts/script_sys_gamesys.cpp

```diff
@@ -0,0 +1,476 @@
+     * any matching entries found. If not, the path will be tried
+     * as is from the primary disk on the device.
+     *
+     * In order for the engine to include custom resources in the build process, you need
+     * to specify them in the "custom_resources" key in your "game.project" settings file.
+     * You can specify single resource files or directories. If a directory is included
+     * in the resource list, all files and directories in that directory is recursively
+     * included:
+     *
+     * For example "main/data/,assets/level_data.json".
+     *
+     * Note that issuing multiple requests of the same resource will yield
+     * individual buffers per request. There is no implic caching of the buffers
+     * based on request path.
+     *
+     * @name sys.load_buffer_async
+     * @param path [type:string] the path to load the buffer from
+     * @param status_callback [type:function(self, request_id, status, buf)] A status callback that will be invoked when a request has been handled, or an error occured
+     * @return handle [type:handle] a handle to the request
+     * @examples
+     *
+     * Load binary data from a custom project resource and update a texture resource:
+     *
+     * ```lua
+     * function my_callback(self, request_id, status, buf)
+     *   resource.set_texture("/my_texture", { ... }, buf)
+     * end
+     *
+     * local my_request = sys.load_buffer_async("/assets/my_level_data.bin", my_callback)
+     * ```
+     *
+     * Load binary data from non-custom resource files on disk:
+     *
+     * ```lua
+     * function my_callback(self, request_id, status, buf)
+     *   if status ~= sys.REQUEST_STATUS_FINISHED then
+     *     -- uh oh! File could not be found, do something graceful
+     *   elif request_id == self.first_asset then
+     *     -- buffer contains data from my_level_asset.bin
+     *   elif request_id == self.second_asset then
+     *     -- buffer contains data from 'my_level.bin'
+     *   end
+     * end
+     *
+     * function init(self)
+     *   self.first_asset = hash("folder_next_to_binary/my_level_asset.bin")
+     *   self.second_asset = hash("/some_absolute_path/my_level.bin")
+     *   self.first_request = sys.load_buffer_async(self.first_asset, my_callback)
+     *   self.second_request = sys.load_buffer_async(self.second_asset, my_callback)
+     * end
+     * ```
+     */
+    static int Sys_LoadBufferAsync(lua_State* L)
+    {
+        int top          = lua_gettop(L);
+        const char* path = luaL_checkstring(L, 1);
+        dmScript::LuaCallbackInfo* callback_info = dmScript::CreateCallback(dmScript::GetMainThread(L), 2);
+
+        if (callback_info == 0x0)
+        {
+            return luaL_error(L, "sys.load_buffer_async failed to create callback");
+        }
+
+        dmhash_t path_hash = dmHashString64(path);
+        {
+            dmMutex::ScopedLock lk(g_SysModule.m_LoadRequestsMutex);
```

Comment:
Unless you need to interact with the lock, use the define `DM_MUTEX_SCOPED_LOCK`